d2/src/main/java/com/linkedin/d2/xds/XdsChannelFactory.java

```diff
@@ -0,0 +1,27 @@
+package com.linkedin.d2.xds;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class XdsChannelFactory
+{
+  private static final Logger _log = LoggerFactory.getLogger(XdsChannelFactory.class);
+
+  public ManagedChannel createChannel(String xdsServerUri)
+  {
+    if (xdsServerUri == null || xdsServerUri.isEmpty())
+    {
+      _log.error("No xDS server address provided");
+      return null;
+    }
+
+    return ManagedChannelBuilder.forTarget(xdsServerUri)
+        .usePlaintext()
```

Comment:
Add a TODO here that we will switch to use TLS

Reply:
Added